File: clojure/difference-of-squares/src/difference_of_squares.clj
(ns difference-of-squares)

(defn square-of-sum
  "Returns the square of the sum of the numbers up to N."
  [N]
  (let [sum (reduce + (range (inc N)))] (* sum sum)))

(defn sum-of-squares
  "Returns the sum of the squares of the numbers up to N."
  [N]
  (reduce + (map #(* % %) (range (inc N)))))

(defn difference
  "Returns the difference between the square of the sum
  and the sum of the squares of the numbers up to N."
  [N]
  (- (square-of-sum N) (sum-of-squares N)))

File: python/difference-of-squares/difference_of_squares.py
"""Functions for sum of squares, square of sums, and difference"""

def square_of_sum(number):
    return sum(num for num in range(1, number + 1)) ** 2


def sum_of_squares(number):
    return sum(num ** 2 for num in range(1, number + 1))


def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)
